x-pack/test/functional/services/monitoring/logstash_pipeline_viewer.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export function MonitoringLogstashPipelineViewerProvider({ getService }) {
+  const testSubjects = getService('testSubjects');
+  const retry = getService('retry');
+  const find = getService('find');
+
+  const PIPELINE_VIEWER_SELECTOR = '.monPipelineViewer';
+  const SUBJ_PIPELINE_SECTION_PREFIX = 'pipelineViewerSection_';
+  const PIPELINE_SECTION_ITEM_CLS = 'monPipelineViewer__listItem';
+
+  return new (class LogstashPipelineViewer {
```

Comment:
The return new class pattern is really interesting. Not sure why we do it that way, but indeed this fits with the rest of the monitoring test services.